backend/src/main/kotlin/cz/loono/backend/api/service/HealthcareProvidersService.kt

```diff
@@ -1,40 +1,143 @@
 package cz.loono.backend.api.service
 
-import cz.loono.backend.api.UpdateStatusMessage
+import com.google.gson.Gson
+import cz.loono.backend.api.dto.HealthcareProviderDetailsDto
+import cz.loono.backend.api.dto.HealthcareProviderIdDto
+import cz.loono.backend.api.dto.HealthcareProviderListDto
+import cz.loono.backend.api.dto.SimpleHealthcareProviderDto
+import cz.loono.backend.api.dto.UpdateStatusMessageDto
 import cz.loono.backend.api.exception.LoonoBackendException
 import cz.loono.backend.data.HealthcareCSVParser
 import cz.loono.backend.data.constants.CategoryValues
 import cz.loono.backend.data.constants.Constants.OPEN_DATA_URL
 import cz.loono.backend.db.model.HealthcareCategory
+import cz.loono.backend.db.model.HealthcareProvider
 import cz.loono.backend.db.repository.HealthcareCategoryRepository
 import cz.loono.backend.db.repository.HealthcareProviderRepository
+import io.github.reactivecircus.cache4k.Cache
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.http.HttpStatus
+import org.springframework.scheduling.annotation.Scheduled
 import org.springframework.stereotype.Service
 import org.springframework.transaction.annotation.Transactional
+import java.io.ByteArrayOutputStream
+import java.io.IOException
 import java.net.URL
+import java.util.zip.ZipEntry
+import java.util.zip.ZipOutputStream
 
 @Service
 class HealthcareProvidersService @Autowired constructor(
     private val healthcareProviderRepository: HealthcareProviderRepository,
     private val healthcareCategoryRepository: HealthcareCategoryRepository
 ) {
 
+    private val providersCache = Cache.Builder().build<HealthcareProviderIdDto, HealthcareProvider>()
+    private val fileCache = Cache.Builder().build<String, ByteArray>()
+
+    @Scheduled(cron = "0 0 2 1 * ?") // each the 1st day of month at 2AM
```

Comment:
Nie som istý, či tam už budú aktuálne dáta. Díval si sa, kedy to UZIS zverejňuje? Za mňa radšej to dať druhý deň v mesiaci s tým, že budem mať deň staré dáta ako riskovať, že ich budem mať staré 30 dní.